uPortal-layout/uPortal-layout-impl/src/main/java/org/apereo/portal/layout/dlm/PositionManager.java

```diff
@@ -180,12 +181,24 @@ static void adjustPositionSet(
         }
 
         // now for any left over after the insert point remove them.
-
-        while (nodeToInsertBefore.getNextSibling() != null)
-            positionSet.removeChild(nodeToInsertBefore.getNextSibling());
+        int removeFails = 0; // avoid infinite loop, if stuck
+        while (nodeToInsertBefore.getNextSibling() != null && removeFails < 5) {
+            Node toRemove = nodeToInsertBefore.getNextSibling();
+            try {
+                toRemove.getParentNode().removeChild(toRemove);
```

Comment:
Should `removeFails` reset on successful removal?

Reply:
Good question. I wanted to make sure no more than 5 total removal errors occur during the cleanup loop